rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/impl/MediaTypeHeaderProvider.java

```diff
@@ -190,4 +190,35 @@ private static MediaType handleMediaTypeWithoutSubtype(String mType) {
         }
         throw new IllegalArgumentException("Media type separator is missing");
     }
+
+    // Determines whether the type or subtype contains any of the tspecials characters defined at:
+    // https://tools.ietf.org/html/rfc2045#section-5.1
+    private static boolean isValid(String str) {
+        final int len = str.length();
```

Comment:
Minor, may be check `str.trim().length()`, it seems weird media types like `application/   ` are still passing through.

Reply:
@reta - sounds good.  Since we end up trimming both type and subType at lines 114 and 115, I moved those trims to before the `isValid` method is called (to lines 88 and 89).  Let me know if you'd prefer the trim to be in isValid instead.  Otherwise, I'll plan to merge this later today.  Thanks for the review!

Reply:
Looks good, thanks @andymc12 !